### ISSUE TITLE
Clarify DDP activation checkpointing support

### ISSUE THREAD

**Issue by rohan-varma** [pytorch/pytorch#68781]
## 📚 Documentation

DDP offers some limited support for working with gradient checkpointing, in situations where module is checkpointed once or twice, with or without shared weights, and different behaviors depending on whether find_unused_parameters=True or not, static_graph=True or not.

This is quite confusing for users and there is no centralized documentation to clearly indicate what is supported and what is not. We should add this to the PT DDP docs: https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html

**Comment by rohan-varma** [comment]
We can likely take information from https://github.com/pytorch/pytorch/pull/52894 to populate these docs.
